hack/.clusterUtils

```diff
@@ -0,0 +1,47 @@
+# shellcheck shell=bash
+
+makeSecretForCluster() {
+  local clusterName=$1
+
+  local tmpfile=$(mktemp /tmp/kubeconfig-internal.XXXXXX)
+  ${KIND_BIN} export kubeconfig -q --internal --name ${clusterName} --kubeconfig ${tmpfile}
+  local server=$(kubectl --kubeconfig ${tmpfile} config view -o jsonpath="{$.clusters[?(@.name == 'kind-${clusterName}')].cluster.server}")
+  local caData=$(kubectl --kubeconfig ${tmpfile} config view --raw -o jsonpath="{$.clusters[?(@.name == 'kind-${clusterName}')].cluster.certificate-authority-data}")
+  local certData=$(kubectl --kubeconfig ${tmpfile} config view --raw -o jsonpath="{$.users[?(@.name == 'kind-${clusterName}')].user.client-certificate-data}")
+  local keyData=$(kubectl --kubeconfig ${tmpfile} config view --raw -o jsonpath="{$.users[?(@.name == 'kind-${clusterName}')].user.client-key-data}")
+  rm -f ${tmpfile}
+
+  cat <<EOF
+kind: Secret
+apiVersion: v1
+metadata:
+  name: ""
+  namespace: ""
+stringData:
+  config: >-
+    {
+      "tlsClientConfig":
+        {
+          "insecure": false,
+          "caData": "${caData}",
+          "certData": "${certData}",
+          "keyData": "${keyData}"
+        }
+    }
+  name: ${clusterName}
```

Comment:
it looks like this value will be "mctc-control-plane" but it should be the name of the workload cluster it's being put into (e.g. "mctc-workload-1").

Reply:
Hmm that is not as simple to automate as I thought it would be, since it means the secret will change for every cluster that it's created for. I think the only way to automate it will be to take it from the context or the kubeconfig. Any ideas?

Reply:
in my PR, I updated the logic in `#8. Add workload clusters if MCTC_WORKLOAD_CLUSTERS_COUNT environment variable is set` to create the secret on each workload cluster, maybe this could do the same?

Reply:
This is kind of related to the issues i was thinking kustomize would help us with, or at least we should be generating a set of k8s resources per cluster, that can be applied directly using `kustomize build x | kubectl apply -f`. Probably using an overlay per cluster that uses base resources in our config dir.

I'd say for now though, if we can just get this working, we can look at something like that later.

Reply:
I expect to open my PR today, if you like we can just go with my solution. It's also not ideal, but saves you the effort of changing this PR @sergioifg94

Reply:
I'm following your suggestion for now @philbrookes and adding the secret in the local-setup. I'm testing it and I'll update the PR when I'm sure it works

Reply:
Just pushed the changes. The secret is now created automatically in the local-setup with the correct name